sinatra-contrib/lib/sinatra/namespace.rb

```diff
@@ -284,8 +284,8 @@ def respond_to(*args)
       end
 
       def set(key, value = self, &block)
+        return key.each { |k,v| set(k, v) } if key.is_a?(Enumerable) and block.nil? and value == self
```

Comment:
Thanks for taking a look at this @BuonOmo
What do you think about using each_pair instead of each?

Reply:
Hi @namusyaka 

If it feels important, please feel free to change it. I wouldn't since one may want to write something like:

```ruby
set { foo: 1, bar: 5 }.map { |k, v| [k + "s", v.to_s] }
```